week-2/interpret/time-format.js

```diff
@@ -12,32 +12,32 @@ function formatTimeDisplay(seconds) {
   const totalHours = (totalMinutes - remainingMinutes) / 60;
   const remainingHours = totalHours % 24;
 
-  return `${pad(remainingHours)}:${pad(remainingMinutes)}:${pad(
-    remainingSeconds
-  )}`;
+  return `${remainingHours.toString().padStart(2, "0")}:${remainingMinutes
+    .toString()
+    .padStart(2, "0")}:${remainingSeconds.toString().padStart(2, "0")}`;
 }
 
-console.log(formatTimeDisplay(143));
+console.log(formatTimeDisplay(323485723465));
 
 // You can play computer with this example
 // Use the Python Visualiser https://pythontutor.com/visualize.html#mode=edit
 // to help you answer these questions
 
 // Questions
 
-// a) When formatTimeDisplay is called how many times will pad be called?
+// a) When formatTimeDisplay is called how many times will pad be called?                      3 times
 
 // Call formatTimeDisplay with an input of 143, now answer the following:
 
-// b) What value is assigned to the parameter num when pad is called for the first time?
+// b) What value is assigned to the parameter num when pad is called for the first time?       143 assigned to parameter
 
-// c) What is the return value of pad when it is called for the first time?
+// c) What is the return value of pad when it is called for the first time?                    00
 
-// d) What is the value assigned to the parameter num when pad
+// d) What is the value assigned to the parameter num when pad                                 23, because remaining seconds is the remainder of total seconds when divided by 60
 // is called for the last time in this program?  Explain your answer
 
-// e) What is the return value when pad is called
+// e) What is the return value when pad is called                                              23, because 23 is bigger than 10 so pad returns num itself
 // for the last time in this program?  Explain your answer
 
-// f) Research an alternative way of padding the numbers in this code.
+// f) Research an alternative way of padding the numbers in this code.                         we can use .padStart(2, "0") it gives us min 2 characters with filling "0"
```

Comment:
This is a perfect alternative way to give the padding we want. Firstly converting the number-type data to the string-type data and then using .padStart() method! 👍